azurelinuxagent/common/protocol/wire.py

```diff
@@ -1195,14 +1196,78 @@ def get_artifacts_profile(self):
                     msg = "Content: [{0}]".format(profile)
                     logger.verbose(msg)
 
-                    from azurelinuxagent.common.event import report_event, WALAEventOperation
                     report_event(op=WALAEventOperation.ArtifactsProfileBlob,
                                  is_success=False,
                                  message=msg,
                                  log_event=False)
 
         return artifacts_profile
 
+    def upload_logs(self, content):
+        try:
+            host_plugin = self.get_host_plugin()
+            host_plugin.put_vm_log(content)
+
+            msg = "Upload VM logs request succeeded using the host plugin channel with " \
+                  "container id {0} and role config file {1}".format(host_plugin.container_id,
+                                                                     host_plugin.role_config_name)
+            logger.info(msg)
+            add_event(name=AGENT_NAME,
+                      version=CURRENT_VERSION,
+                      op=WALAEventOperation.HostPlugin,
+                      is_success=True,
+                      message=msg,
+                      log_event=False)
+            return
+        except (ResourceGoneError, InvalidContainerError) as e:
+            host_plugin = self.get_host_plugin()
+            old_container_id = host_plugin.container_id
+            old_role_config_name = host_plugin.role_config_name
+
+            msg = "Upload VM logs request failed with the current host plugin configuration. " \
+                  "ContainerId: {0}, role config file: {1}. Fetching new goal state and retrying the call." \
+                  "Error: {2}".format(old_container_id, old_role_config_name, ustr(e))
+            logger.info(msg)
+
+            self.update_host_plugin_from_goal_state()
+
+            new_container_id = host_plugin.container_id
+            new_role_config_name = host_plugin.role_config_name
+            msg = "Host plugin reconfigured with new parameters. " \
+                  "ContainerId: {0}, role config file: {1}.".format(new_container_id, new_role_config_name)
+            logger.info(msg)
+
+            try:
+                host = self.get_host_plugin()
```

Comment:
just curious: why do we keep calling self.get_host_plugin() and why use a different variable here? (multiple calls are ok, but the use of a new variable threw me off)

Reply:
This is outdated now. I didn't have a reason why I was doing it like that.